src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleArray.cs

```diff
@@ -261,6 +271,23 @@ public static partial class JsonSerializer
                     ThrowHelper.ThrowJsonException_DeserializeDuplicateKey(key, reader, state.PropertyPath);
                 }
             }
+            else if (state.Current.IsDictionaryProperty)
+            {
+                Debug.Assert(state.Current.JsonPropertyInfo != null);
+                Debug.Assert(state.Current.ReturnValue != null);
+                IDictionary<string, TProperty> dictionary = (IDictionary<string, TProperty>)state.Current.JsonPropertyInfo.GetValueAsObject(state.Current.ReturnValue);
+
+                string key = state.Current.KeyName;
+                Debug.Assert(!string.IsNullOrEmpty(key));
+                if (!dictionary.ContainsKey(key)) // The IDictionary.TryAdd extension method is not available in netstandard.
```

Comment:
What is going on here? Is this for deserializing a dictionary?

I would recommend "last value" wins when setting values in a dictionary that has multiple keys. If you want to be strict then there should at least be a setting to opt out.

Reply:
Yes - any dictionary whether from extension data or not. It is used to take a return value from the manual stack-based approach (such as an object that needs to be added to a dictionary) and add it. It is somewhat of a barnacle - I'm considering a lambda\delegate to do this so the caller specifies the right semantics instead of the `if` statement and stack-based state.